eureka2-client/src/main/java/com/netflix/eureka2/client/interest/EurekaInterestClientImpl.java

```diff
@@ -71,54 +57,8 @@ public EurekaInterestClientImpl(SourcedEurekaRegistry<InstanceInfo> registry,
 
         this.retryableConnection = retryableConnectionFactory.singleOpConnection(opStream, executeOnChannel);
 
-        // subscribe to the base interest channels to do cleanup on every channel refresh.
-        retryableConnection.getChannelObservable()
```

Comment:
Unfortunately we cannot do that, as ```retryableConnection``` is not constructed yet. But as these methods are protected, the risk of misuse is not big.